Modules/SubModule_Import/xASL_imp_BIDS2Legacy.m

```diff
@@ -50,39 +50,37 @@
     
     [rawdataFolder, fileNameDatasetDescription, fileExtension] = fileparts(x.dir.dataset_description);
     if strcmpi([fileNameDatasetDescription fileExtension], 'dataset_description.json')
-        %% 2.1 The input is dataset_description.json in the rawdata folder
+        % 2.1 The input is dataset_description.json in the rawdata folder
         [~, rawdataFolderName] = fileparts(rawdataFolder);
         if ~strcmp(rawdataFolderName, 'rawdata')
             error('Invalid folder in which dataset_description.json was found, should be /rawdata');
         end
-        localDatasetRoot = x.dir.DatasetRoot;
     else
-        %% 2.2 The input is dataPar.json or sourceStructure.json - have to look for a rawdata folder
+        % 2.2 The input is dataPar.json or sourceStructure.json - have to look for a rawdata folder
         pathRawData = fullfile(x.dir.DatasetRoot,'rawdata');
         % Check the if the correct BIDS structure is in place
         if ~exist(pathRawData,'dir')
             error('Path rawdata does not exist');
         elseif ~exist(fullfile(pathRawData,'dataset_description.json'),'file')
             error('File dataset_description.json is not found');
-        else
-            localDatasetRoot = x.dir.DatasetRoot;
         end
     end
 
 	%% 3. Run the legacy conversion: Check if a dataPar is provided, otherwise use the defaults
-	fListDataPar = xASL_adm_GetFileList(localDatasetRoot,'(?i)(^dataPar.*\.json$)', 'FPList', [], 0);
-	if length(fListDataPar) < 1
-		fprintf('There is no dataPar.json file in the study root directory. Default settings will be used...\n');
-		% Fill the dataPars with default parameters
-		x = xASL_bids_BIDS2Legacy(localDatasetRoot, x, 1);
-	else
-		if length(fListDataPar) > 1
-			warning('Multiple dataPar*.jsons exist. Using the first: %s\n',fListDataPar{1});
-		end
-		% Fill the dataPars with the provided parameters
-		x.dataPar = spm_jsonread(fListDataPar{1});
-		x = xASL_bids_BIDS2Legacy(localDatasetRoot, x, 1);
+	fListDataPar = xASL_adm_GetFileList(x.dir.DatasetRoot,'(?i)(^dataPar.*\.json$)', 'FPList', [], 0);
+    if length(fListDataPar) < 1
+        % Fill the dataPars with default parameters
+        fprintf('There is no dataPar.json file in the study root directory. Default settings will be used...\n');
```

Comment:
What happens if, at the start of the whole import, only `/DatasetRoot/derivatives/ExploreASL/dataPar.json` exists? That would be the first thing a user would try right? At least this shouldn't be overwritten. Perhaps we can detect this, copy to the datasetRoot, and issue a warning that this was found and will be used.

Reply:
The workflow already checks for dataPar.json files in derivatives. I added a print statement for you 2fc744d

Reply:
I would solve this differently. `dataPar.json` contains settings for processing, and you recommend the user to put this in `/datasetRoot`, and then at the end of the `BIDS2legacy` submodule, this `dataPar.json` will be merged with the new fields that are generated automatically, if these fields didn't exist (e.g., `x.Quality`, `x.SUBJECTS` etc). At least, this should happen!

My point here was that, if the user created a `/datasetRoot/derivatives/ExploreASL/dataPar.json` instead of `/datasetRoot/dataPar.json`, you move the `/datasetRoot/derivatives/ExploreASL/dataPar.json` to `/datasetRoot/dataPar.json`, so that in the end of the `BIDS2legacy` submodule, they will be merged appropriately.

If the user reruns ExploreASL and forgets to delete the previous `/datasetRoot/derivatives/ExploreASL/dataPar.json`, this is the user's fault (or we could warn the user at the beginning of all modules everything, if no lock files exist yet, if you want).